Mathlib/RingTheory/FractionalIdeal/Extended.lean

```diff
@@ -209,6 +209,52 @@ theorem coe_extendedHomₐ_eq_span (I : FractionalIdeal A⁰ K) :
     IsLocalization.algebraMap_eq_map_map_submonoid A⁰ B K L]
   rfl
 
+variable [IsIntegrallyClosed A] [IsIntegrallyClosed B]
+
+theorem le_one_of_extendedHomₐ_le_one (hI : extendedHomₐ L B I ≤ 1) : I ≤ 1 := by
+  contrapose! hI
+  rw [SetLike.not_le_iff_exists] at hI ⊢
+  obtain ⟨x, hx₁, hx₂⟩ := hI
+  refine ⟨algebraMap K L x, ?_, ?_⟩
+  · simpa [← FractionalIdeal.mem_coe, IsLocalization.algebraMap_eq_map_map_submonoid A⁰ B K L]
+      using subset_span <| Set.mem_image_of_mem _ hx₁
+  · contrapose! hx₂
+    rw [mem_one_iff, ← IsIntegrallyClosed.isIntegral_iff] at hx₂ ⊢
+    exact IsIntegral.tower_bot_of_field <| isIntegral_trans _ hx₂
+
+theorem extendedHomₐ_le_one_iff : extendedHomₐ L B I ≤ 1 ↔ I ≤ 1 :=
+  ⟨fun h ↦ le_one_of_extendedHomₐ_le_one L B h, fun a ↦ extended_le_one_of_le_one L _ I a⟩
+
+variable [IsDedekindDomain A] [IsDedekindDomain B]
```

Comment:
Any lemma below will get `[IsDedekindDomain A] [IsIntegrallyClosed A]` in their assumption right?
Can you put this part into a section to avoid this? Thanks.